### ISSUE TITLE
Make the Collectors returned by Collectors2AdditionalTest.sumBy*() methods thread-safe.

### ISSUE THREAD

**Issue by motlin** [eclipse/eclipse-collections#1680]
In #1679, we deleted flaky tests that were using these Collectors as if they were thread-safe from a parallel test. It's not clear to me if all Collectors must be thread-safe. Currently they are not thread-safe because they delegate directly to hashtable mutation methods on a non-thread-safe map. If these Collectors will be used from parallel contexts, we should make them thread-safe and possibly restore the deleted tests.

**Comment by Riya-Sharma12** [comment]
Hi @motlin !

I'm Riya Sharma, and I'm new to contributing to open source. I came across this issue and would love to help solve it. I’ve read through the description and think it would be a great opportunity to start contributing.

Looking forward to contributing and making a positive impact here!

**Comment by vaibhaaavvv** [comment]
Hi @motlin. I have looked into this issue and think I have come up with a fix but cannot run a local test on my fixes. I am not able to build my local repository. My issue: Not able to import any org.eclipse.collections*. Could you please help me set up this repository so that I can raise a PR?
Thanks in advance.

**Comment by motlin** [comment]
@vaibhaaavvv have you followed the steps in https://github.com/eclipse/eclipse-collections/blob/master/CONTRIBUTING.md? If so, what step fails, and with what error message?

**Comment by vaibhaaavvv** [comment]
Yes @motlin I followed these steps and I am getting a compilation error in all the import org.eclipse.collections.api.block.function.primitive.*
<img width="647" alt="image" src="https://github.com/user-attachments/assets/056f5826-8da0-430a-81bb-1a13cee2ff25">

**Comment by motlin** [comment]
@vaibhaaavvv those look like IDE errors. Do you get any errors when you run the maven build?

**Comment by vaibhaaavvv** [comment]
No. I am able to build the project successfully before importing the repo on my IDE.

**Comment by motlin** [comment]
Sounds like a problem where the IDE doesn't recognize all the generated sources folders. I think you're using the Eclipse IDE, is that right?. Most of the contributors here use IntelliJ. Are you familiar with how to configure extra generated sources folder?

**Comment by vaibhaaavvv** [comment]
Yes. In this case I am using Eclipse IDE. I'll try to import it in Intellij and will let you know if I face any issue.
